src/pages/index.tsx

```diff
@@ -4,19 +4,108 @@ import { Layout } from "../components/Layout";
 import { useRouter } from "next/router";
 import dynamic from "next/dynamic";
 import { Posts } from "../components/Post";
+import { api } from "../utils/api";
+import { useEffect, useState } from "react";
+import { BlogPostCategory } from "../types/blogPost";
+import SearchPost from "../components/SearchPost";
+import { usePagination } from "../hooks/usePagination";
+import { useDebounce } from "../hooks/useDebounce";
+import { Pagination } from "../components/Pagination";
 
 const HomePage: NextPage = () => {
+  const [paginationPage, setPaginationPage] = useState(0);
+  const [searchActive, SetSearchActive] = useState(false);
+  const [searchCategory, setSearchCategory] = useState<BlogPostCategory[]>([]);
+  const [searchWord, setSearchWord] = useState<string>("");
+  const debounce = useDebounce(searchWord, 500);
```

Comment:
You can debounce whole filters, so

```
const debouncedPostListFilters = useDebouce(postListFilters, 500);
```

500 should be moved to const above page component `const POST_LIST_FILTERS_DEBOUNCE = 500`